designs/instance-type-settings.md

```diff
@@ -0,0 +1,340 @@
+
+### Custom Resource Cardinality
+
+One of the primary questions presented is whether we should create a 1-1 mapping between settings and a given instance type or whether we should allow the ability to create more complex structures using set relationships and label selectors. Options are listed below combined with the pros and cons for each:
+
+**Options**
+
+1. Create an `InstanceType` CRD that specifies a 1-1 relationship between the settings and the instance type
+    1. Pros
+        1. Clearly defined relationship between an instance type and the settings that are defined for it
+        2. Users that only have a few instance types will not have an issue configuring the instance types with these values
+        3. We can consider adding an `InstanceTypeClass` or similar at a later date that would allow a grouping of instance types to be assigned the same settings
+    2. Cons
+        1. May not scale well as users that have high levels of customizations across large numbers of instance types may have to maintain a large number of `InstanceType` CRDs
+        2. I may want to define settings over an instance family without having to individually configure the instances that exist within that family
+2. Create an `InstanceTypeSetting` CRD that contains a list of instance types that allow a 1-many relationship between settings and instance types. These instance types could be selected either with a `selector` or `regex`
+    1. Pros
+        1. Allows users who want to apply the same instance type settings across a wide range of instance types or across an instance type family to do so
+        2. Lower maintenance burden from maintaining less `InstanceTypeSetting` CRDs
+    2. Cons
+        1. Can create complex relationships that may be difficult to track for users
+        2. If we were to allow overlapping `InstanceTypeSettings`, we would have to create a `weight` on the `InstanceTypeSettings` which would further convolute details for the user
+        3. If we were to deny overlapping `InstanceTypeSettings` , then the feature of specifying a group of instance types may not be used that often as configuration between instance types may rarely be shared
+        4. It’s unclear whether users would want have enough common configuration that they would assign across instance types. It makes more sense to iterate on a grouping mechanism later if there is a user ask for it
+
+**Recommendation:** Use `InstanceType` CRD for initial iteration for 1-1 mappings before considering introducing more complex constructs to the user. While specifying rulesets might reduce the overhead burden for customers to keep track of separate `InstanceTypeSettings` , ultimately without an API surface to view exactly what settings that ruleset will produce, it will ultimately be confusing for users.
```

Comment:
```suggestion
**Recommendation:** Use `InstanceType` CRD for initial iteration for one-to-one mappings before considering introducing more complex constructs to the user. While specifying rulesets might reduce the overhead burden for customers to keep track of separate `InstanceTypeSettings`, ultimately without an API surface to view exactly what settings that ruleset will produce, it will ultimately be confusing for users.
```